### PR TITLE
Provide a configurable weight value for MSTransferor Rucio rules

### LINKED ISSUE #10617: [MS] Create input data placement rules with a configurable weight expression
**Impact of the new feature**
MSTransferor

**Is your feature request related to a problem? Please describe.**
Yes, in addition to the extra complexity that we need to introduce to the WMCore microservice code, it can also result in non-ideal and unbalanced data distribution.

**Describe the solution you'd like**
We should start providing a `weight` expression to the rucio creation call. The initial idea would be to use `ddm_quota` as the weight expression, but this is likely going to change in the coming weeks/months, so we should make this weight expression configurable via MS configuration.

Thus, changes to both service configuration and WMCore source code are required here.

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #10617 

#### Status
not-tested

#### Description
With this PR, we start creating input data placement Rucio rules with a default weight expression:
```
weight="ddm_quota"
```

this weight expression has been made configurable through the service configuration though.

#### Is it backward compatible (if not, which system it affects?)
yes

#### Related PRs
None

#### External dependencies / deployment changes
It requires the following configuration/deployment changes:
https://github.com/dmwm/deployment/pull/1073

Note that it contains also a couple of MSUnmerged configuration changes, as discussed with Todor.

### CODE REVIEW

**Comment by amaltaro** [comment]
test this please

File: src/python/WMCore/MicroService/MSTransferor/MSTransferor.py [hunk 1]
```diff
@@ -670,20 +672,22 @@ def makeTransferRucio(self, wflow, dataIn, subLevel, blocks, dataSize, nodes, no
...
-                'comment': 'WMCore MSTransferor input data placement'}
+        ruleAttrs = {'copies': 1,
+                     'activity': 'Production Input',
+                     'lifetime': self.msConfig['rulesLifetime'],
+                     'account': self.msConfig['rucioAccount'],
+                     'grouping': subLevel,
+                     'meta': {'workflow_group': wflow.getWorkflowGroup()},
+                     'comment': 'WMCore MSTransferor input data placement'}
+        # add a configurable weight value
+        ruleAttrs["weight"] = self.msConfig['rucioRuleWeight']
```
Comment (todor-ivanov) [inline src/python/WMCore/MicroService/MSTransferor/MSTransferor.py:683]: Why not in the `ruleAttrs={}` definition itself?

**Review by todor-ivanov — APPROVED** [review]
@amaltaro,  I left one minor comment inline, but do not feel like changing it. It will work perfectly fine as it is right now. So I think the change you suggest here is looking already good enough.

File: src/python/WMCore/MicroService/MSTransferor/MSTransferor.py (hunk 1 above)
Comment (amaltaro) [reply src/python/WMCore/MicroService/MSTransferor/MSTransferor.py:683]: Good question! Let me quickly update it. Thanks

**Comment by amaltaro** [comment]
Thanks for the prompt review, Todor.

**Pull request merged**